Shipr-Helm/templates/shipr-frontend-deployment.yaml

```diff
@@ -0,0 +1,53 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ Values.frontend.name }}
+spec:
+  replicas: {{ Values.frontend.replicas }}
+  selector:
+    matchLabels:
+      app: {{ Values.frontend.name }}
+  template:
+    metadata:
+      labels:
+        {{- with .Values.frontend.labels }}
+        {{- toYaml . | nindent 6 }}
+        {{- end }}
+    spec:
+      containers:
+      - name: {{ Values.frontend.name }}
+        image: {{ Values.frontend.repository }}:{{ Values.frontend.tag }}
+        ports:
+        - containerPort: {{ Values.frontend.containerPort }}
+        livenessProbe:
+          httpGet:
+            path: /
+            port: {{ Values.frontend.containerPort }}
+          initialDelaySeconds: 10
+          periodSeconds: 10
+        readinessProbe:
+          httpGet:
+            path: /
+            port: {{ Values.frontend.containerPort }}
+          initialDelaySeconds: 5
+          periodSeconds: 5
+        startupProbe:
+          httpGet:
+            path: /
+            port: {{ Values.frontend.containerPort }}
+          initialDelaySeconds: 15
+          periodSeconds: 5
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ Values.frontend.name }}
+spec:
+  selector:
+    app: {{ Values.frontend.name }}
+  ports:
+    - protocol: TCP
+      port: {{ Values.frontend.service.port }}
+      targetPort: {{ Values.frontend.service.targetport }}
+      nodePort: {{ Values.frontend.service.nodePort }}
+  type: NodePort
```

Comment:
Again, what if clusterIP type is needed here?

Reply:
Used dynamic value